solana-programs/anchor/audius-data/programs/audius-data/src/lib.rs

```diff
@@ -95,10 +103,250 @@ pub mod audius_data {
 
         let audius_user_acct = &mut ctx.accounts.user;
         audius_user_acct.eth_address = eth_address;
+        audius_user_acct.ursm = ursm;
+
+        Ok(())
+    }
+
+    /// Create a content node account from the admin account.
+    /// The content node's account is derived from the admin base PDA + sp_id bytes.
+    pub fn create_content_node(
+        ctx: Context<CreateContentNode>,
+        base: Pubkey,
+        sp_id: u16,
+        authority: Pubkey,
+        owner_eth_address: [u8; 20],
+    ) -> Result<()> {
+        msg!("Audius::CreateContentNode");
+
+        // Confirm that the base used for content node account seed is derived from this Audius admin storage account
+        let (derived_base, _) = Pubkey::find_program_address(
+            &[&ctx.accounts.admin.key().to_bytes()[..32]],
+            ctx.program_id,
+        );
+
+        if derived_base != base {
+            return Err(ErrorCode::Unauthorized.into());
+        }
+
+        // Confirm that the derived pda from base is the same as the user storage account
+        let (derived_content_node, _) = Pubkey::find_program_address(
+            &[&derived_base.to_bytes()[..32], CONTENT_NODE_SEED_PREFIX, sp_id.to_le_bytes().as_ref()],
```

Comment:
why is CONTENT_NODE_SEED_PREFIX needed if it's just a constant? does it make a difference?

Reply:
nvm! i saw we do it as a good practice to differentiate account types.